ckanext/validation/utils.py

```diff
@@ -37,14 +37,22 @@ def process_schema_fields(data_dict):
     2. If `schema_url` is provided and looks like a valid URL, it's copied
         to `schema`
     3. If `schema_json` is provided, it's copied to `schema`.
+    4. If `schema_json` is provided but blank, the schema is deleted.
 
     All the 3 `schema_*` fields are removed from the data_dict.
     Note that the data_dict still needs to pass validation
     '''
 
+    # Distinguish blank field (likely web UI) from absent field (likely API).
+    # Absent fields do not overwrite the existing schema,
+    # but fields that are present and blank can.
+    has_schema_json = u'schema_json' in data_dict
+
     schema_upload = data_dict.pop(u'schema_upload', None)
     schema_url = data_dict.pop(u'schema_url', None)
     schema_json = data_dict.pop(u'schema_json', None)
+    if isinstance(schema_json, str):
```

Comment:
can't you place this inside the elif has_schema_json logic,
and if stripped and none, don't set?

Reply:
But we specifically _want_ to assign it if it strips to `''`, because that indicates that the user is trying to delete it.